snapcraft/internal/repo/snaps.py

```diff
@@ -0,0 +1,210 @@
+        self._store_snap_info = None
+
+        self._is_installed = None
+        self._is_on_store = None
+
+    @property
+    def installed(self):
+        if self._is_installed is None:
+            self._is_installed = self.get_local_snap_info() is not None
+        return self._is_installed
+
+    @property
+    def on_store(self):
+        if self._is_on_store is None:
+            self._is_on_store = self.get_store_snap_info() is not None
+        return self._is_on_store
+
+    def get_local_snap_info(self):
+        """Returns a local payload for the snap.
+
+        Validity of the results are determined by checking self.installed."""
+        if self._is_installed is None:
+            with contextlib.suppress(exceptions.HTTPError):
+                self._local_snap_info = _get_local_snap_info(self.name)
+        return self._local_snap_info
+
+    def get_store_snap_info(self):
+        """Returns a store payload for the snap.
+
+        Validity of the results are determined by checking self.remote."""
+        if self._is_on_store is None:
+            with contextlib.suppress(exceptions.HTTPError):
+                self._store_snap_info = _get_store_snap_info(self.name)
+        return self._store_snap_info
+
+    def _get_store_channels(self):
+        snap_store_info = self.get_store_snap_info()
+        if not self.on_store:
+            return dict()
+
+        return snap_store_info['channels']
+
+    def get_current_channel(self):
+        current_channel = ''
+        if self.installed:
+            local_snap_info = self.get_local_snap_info()
+            current_channel = local_snap_info['channel']
+            if current_channel == 'stable':
+                current_channel = 'latest/stable'
+        return current_channel
+
+    def is_classic(self):
+        store_channels = self._get_store_channels()
+        return store_channels[self.channel]['confinement'] == 'classic'
+
+    def is_valid(self):
+        """Check if the snap is valid."""
+        if not self.on_store:
+            return False
+        store_channels = self._get_store_channels()
+        return self.channel in store_channels.keys()
+
+    def install(self):
+        """Installs the snap onto the system."""
+        snap_install_cmd = []
+        if _snap_command_requires_sudo():
+            snap_install_cmd = ['sudo']
+        snap_install_cmd.extend(['snap', 'install', self.name])
+        if self._original_channel:
+            snap_install_cmd.extend(['--channel', self._original_channel])
+        if self.is_classic:
+            snap_install_cmd.extend(['--classic'])
+        try:
+            check_call(snap_install_cmd)
+        except CalledProcessError as install_error:
+            raise errors.SnapInstallError(snap_name=self.name,
+                                          snap_channel=self.channel)
+
+    def refresh(self):
+        """Refreshes a snap onto a channel on the system."""
```

Comment:
This should be `Refreshes a snap to the latest version, channel and confinement type` because 1) it will update the snap if there's a new version 2) as per my comment on `install_snaps` confinement may have changed.